packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -88,55 +92,187 @@ const fillDerivedPropertyArguments = (
-          genericType,
-          parameter.multiplicity,
-        );
-        if (_type.values.length) {
-          const enumValueRef = EnumValueExplicitReference.create(
-            _type.values[0] as Enum,
+    const querySetupState =
+      derivedPropertyExpressionState.queryBuilderState.querySetupState;
+    let temporalTarget;
+    if (
+      derivedPropertyExpressionState.propertyExpression.func.genericType.value
+        .rawType instanceof Class &&
+      derivedPropertyExpressionState.propertyExpression.func.owner
+        ._generatedMilestonedProperties.length !== 0
+    ) {
+      temporalTarget = getMilestoneTemporalStereotype(
+        derivedPropertyExpressionState.propertyExpression.func.genericType.value
+          .rawType,
+        derivedPropertyExpressionState.queryBuilderState.graphManagerState
+          .graph,
+      );
+    }
+    if (temporalTarget) {
+      switch (temporalTarget) {
+        case MILESTONING_STEROTYPES.BUSINESS_TEMPORAL: {
+          let parameter;
+          if (querySetupState.classMilestoningTemporalValues.length === 0) {
+            derivedPropertyExpressionState.queryBuilderState.buildMilestoningParameter(
+              DEFAULT_MILESTONING_PARAMETERS.BUSINESS_DATE,
+            );
+            parameter = querySetupState.classMilestoningTemporalValues[0];
+          } else if (
+            querySetupState.classMilestoningTemporalValues.length === 1
+          ) {
+            if (
+              querySetupState.classMilestoningTemporalValues[0] instanceof
+                VariableExpression &&
+              querySetupState.classMilestoningTemporalValues[0].name ===
+                DEFAULT_MILESTONING_PARAMETERS.BUSINESS_DATE
+            ) {
+              parameter = querySetupState.classMilestoningTemporalValues[0];
+            } else {
+              derivedPropertyExpressionState.queryBuilderState.buildMilestoningParameter(
+                DEFAULT_MILESTONING_PARAMETERS.BUSINESS_DATE,
+              );
+              parameter = querySetupState.classMilestoningTemporalValues[1];
+            }
+          } else {
+            parameter = querySetupState.classMilestoningTemporalValues[1];
+          }
+          propertyArguments.push(guaranteeNonNullable(parameter));
+          break;
+        }
+        case MILESTONING_STEROTYPES.BITEMPORAL: {
+          if (querySetupState.classMilestoningTemporalValues.length === 0) {
+            derivedPropertyExpressionState.queryBuilderState.buildMilestoningParameter(
+              DEFAULT_MILESTONING_PARAMETERS.PROCESSING_DATE,
+            );
+            derivedPropertyExpressionState.queryBuilderState.buildMilestoningParameter(
+              DEFAULT_MILESTONING_PARAMETERS.BUSINESS_DATE,
+            );
+          } else if (
+            querySetupState.classMilestoningTemporalValues.length === 1
+          ) {
+            if (
+              querySetupState.classMilestoningTemporalValues[0] instanceof
```

Comment:
actually this scenario arises only when parent is not `bitemporal`. If parent is already `bitemporal` `classMilestoningTemporalValues.length` will be 2 and it will just populate the default values.